### ISSUE TITLE
Current perspective name not being propertized in the mode-line

### ISSUE THREAD

**Issue by mukundzare** [nex3/perspective-el#191]
Hi,

I am unable to see the current perspective name being highlighted in the modeline while using ```(setq persp-show-modestring t)```. It highlights correctly in the header when I use ```(setq persp-show-modestring 'header)```. 

If it matters, I am using doom-modeline too. How do I fix this issue?

**Comment by gcv** [comment]
I'm not familiar with doom-modeline. What happens if you disable it?

It could be as simple as a hard-to-see face setting given your theme and modeline configuration. The highlight is controlled using `persp-selected-face` (https://github.com/nex3/perspective-el/blob/cb4083dfa91c336dbed0c330d629e256d86384f9/perspective.el#L500). What happens if you modify it?

**Comment by gcv** [comment]
Since it sounds like everything works with the minimal configuration above, then doom-modeline is _not_ the problem? In that case, yes, I'm inclined to blame some interaction of themes and limitations of `persp-selected-face`, since it's just a `:foreground "Blue"` setting. Also, the face is used directly in the code, rather than through a variable indirection, so it's more difficult to adjust in conflicting situations.

I'd definitely install the conflicting theme and take it from there. It's tedious but not difficult to examine the faces defined by a theme (using `M-x list-faces-display`), and figure out what's going on. The raw mode line is in the `mode-line-format` variable, which I like to examine in `ielm` mode (pretty-printing is important for something that complicated) — it's usually possible to track down which faces are used where by looking at what it's doing.

If this is the case, then maybe a good fix would be to introduce a layer of indirection for `persp-selected-face`, which should make modifying it more theme-friendly:

```
(defface persp-selected-default-face
  '((t (:weight bold :foreground "Blue")))
  "The default face used to highlight the current perspective on the modeline.")

(defvar persp-selected-face persp-selected-default-face
  ;; or better yet: defcustom
  "The face actually used to highlight the current perspective on the modeline.")
```

An even smarter approach is to make the face dynamic, so it automatically adjusts its color to suit the theme. I've done this before for other code, it usually works. A bit more work required, though.

**Comment by mukundzare** [comment]
Yeah, I think hard coding `persp-selected-face` to blue might be causing some interactions with the themes and might need to have an indirection in the case of themes. Besides, I just declared emacs bankruptcy and the themes seemed to have played some part in it or my lack of sufficient knowledge. I have started over with my new config (without perspective.el as of now, but planned in the next few days). So, if the problem reappears, I will report back soon.
